backend/src/services/gitBookService/PageContentManager.ts

```diff
@@ -0,0 +1,50 @@
+import { RequirementStatusEnum } from "myTypes";
+
+type Requirement = {
+    status: RequirementStatusEnum;
+    requirement: string;
+};
+
+class GitBookPageContentManager {
+
+    constructor() {
+    }
+
+    extractRequirements(pageContent): Requirement[] {
+        if (!pageContent || !pageContent.document || !pageContent.document.nodes) {
+            console.log("Invalid page content format.");
+            return [];
+        }
+
+        const nodes = pageContent.document.nodes;
+        const requirements: Requirement[] = [];
+        const numericPrefixRegex = /^\d+(\.\d+)*\s*/;
+
+        nodes.forEach(node => {
+            if (node.type === 'heading-1' && node.nodes) {
+                let textContent = node.nodes.map(n => n.leaves.map(leaf => leaf.text).join('')).join('');
+                textContent = textContent.replace(numericPrefixRegex, ''); // Remove numeric prefix
+
+                let status;
+                if (textContent.includes("(REQUIRED)")) {
+                    status = RequirementStatusEnum.REQUIRED;
+                } else if (textContent.includes("(RECOMMENDED)")) {
+                    status = RequirementStatusEnum.RECOMMENDED;
+                } else if (textContent.includes("(OPTIONAL)")) {
+                    status = RequirementStatusEnum.OPTIONAL;
```

Comment:
could you move these strings to class consts?

Reply:
yes, it would be a good idea to keep all of the const on the top of the class